package/MDAnalysis/coordinates/H5MD.py

```diff
@@ -726,6 +772,670 @@ def __setstate__(self, state):
+            # MPI parallel writing not yet supported
+            self.h5md_file = h5py.File(name=self.filename,
+                                       mode='w',
+                                       driver=self._driver,
+                                       comm=self._comm)
+        elif self._driver is not None:
+            self.h5md_file = h5py.File(name=self.filename,
+                                       mode='w',
+                                       driver=self._driver)
+        else:
+            self.h5md_file = h5py.File(name=self.filename,
+                                       mode='w')
+
+        # fill in H5MD metadata from kwargs
+        self.h5md_file.require_group('h5md')
+        self.h5md_file['h5md'].attrs['version'] = np.array(self.H5MD_VERSION)
+        self.h5md_file['h5md'].require_group('author')
+        self.h5md_file['h5md/author'].attrs['name'] = self.author
+        if self.author_email is not None:
+            self.h5md_file['h5md/author'].attrs['email'] = self.author_email
+        self.h5md_file['h5md'].require_group('creator')
+        self.h5md_file['h5md/creator'].attrs['name'] = self.creator
+        self.h5md_file['h5md/creator'].attrs['version'] = self.creator_version
+
+    def _initialize_hdf5_datasets(self, ts):
+        """initializes all datasets that will be written to by
+        :meth:`_write_next_timestep`
+
+        Note
+        ----
+        :exc:`NoDataError` is raised if no positions, velocities, or forces are
+        found in the input trajectory. While the H5MD standard allows for this
+        case, :class:`H5MDReader` cannot currently read files without at least
+        one of these three groups. A future change to both the reader and
+        writer will allow this case.
+
+
+        """
+
+        # for keeping track of where to write in the dataset
+        self._counter = 0
+
+        # ask the parent file if it has positions, velocities, and forces
+        # if prompted by the writer with the self._write_* attributes
+        self._has = {group: getattr(ts, f'has_{attr}')
+                     if getattr(self, f'_write_{attr}')
+                     else False for group, attr in zip(
+                     ('position', 'velocity', 'force'),
+                     ('positions', 'velocities', 'forces'))}
+
+        # initialize trajectory group
+        self.h5md_file.require_group('particles').require_group('trajectory')
+        self._traj = self.h5md_file['particles/trajectory']
+        self.data_keys = [
+            key for key in ts.data.keys() if key not in self.data_blacklist]
+        if self.data_keys:
+            self._obsv = self.h5md_file.require_group('observables')
+
+        # box group is required for every group in 'particles'
+        self._traj.require_group('box')
+        self._traj['box'].attrs['dimension'] = 3
+        if ts.dimensions is not None and np.all(ts.dimensions > 0):
```

Comment:
Just checking, is the ts.dimensions == 0 as an empty box behaviour also being replicated on read?

Reply:
good call, I added a parameter in the "no dimensions" test and it passes